bluemira/base/components.py

```diff
@@ -158,6 +207,12 @@ def add_children(self, children: List[Component], merge_trees=False):
         self: Component
             This component.
         """
+        if isinstance(children, list) and len(children) == 0:
+            # I'm not sure what the block of code below this is for, but
+            # I assume it does something useful. However, it will throw
+            # an IndexError on an empty list, avoid that here.
+            return self
+
         if not isinstance(children, list) or len(children) == 0:
             child = children[0] if isinstance(children, list) else children
```

Comment:
any chance of a comment here @CoronelBuendia git blame tells me its apparently yours :slightly_smiling_face:?

Probably should narrow it down so we don't say "I dont know what this does" in the code

Reply:
:O That doesn't look like me... I don't normally write 1 list if/elses... but if git blame says so, then I guess I did? If @DanShort12 has no recollection and the tests pass without it, I'd say get rid of anything we don't understand